crates/ruff_linter/src/rules/ruff/rules/missing_await_for_coroutine.rs

```diff
@@ -0,0 +1,140 @@
+use ruff_python_ast::{
+    Expr, ExprAwait, ExprCall, ExprName, Stmt, StmtAssign, StmtExpr, StmtFunctionDef,
+};
+use ruff_text_size::TextRange;
+
+use ruff_diagnostics::{Diagnostic, Edit, Fix, FixAvailability, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_semantic::SemanticModel;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for coroutines that are not awaited. This rule is only active in async contexts.
+///
+/// ## Why is this bad?
+/// Coroutines are not executed until they are awaited. If a coroutine is not awaited, it will
+/// not be executed, and the program will not behave as expected. This is a common mistake when
+/// using `asyncio.sleep` instead of `await asyncio.sleep`.
+///
+/// Python's asyncio runtime will emit a warning when a coroutine is not awaited.
+///
+/// ## Examples
+/// ```python
+/// async def foo():
+///     pass
+///
+///
+/// async def bar():
+///     foo()
+/// ```
+///
+/// Use instead:
+/// ```python
+/// async def foo():
+///     pass
+///
+///
+/// async def bar():
+///     await foo()
+/// ```
+#[violation]
+pub struct MissingAwaitForCoroutine;
+
+impl Violation for MissingAwaitForCoroutine {
+    const FIX_AVAILABILITY: FixAvailability = FixAvailability::Sometimes;
+
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Coroutine is not awaited")
+    }
+
+    fn fix_title(&self) -> Option<String> {
+        Some(format!("Coroutine is not awaited"))
+    }
+}
+
+pub(crate) fn missing_await_for_coroutine(checker: &mut Checker, call: &ExprCall) {
+    // Only check for missing await in async context
+    if !checker.semantic().in_async_context() {
+        return;
+    }
+
+    // Try to detect possible scenarios where await is missing and ignore other cases
+    // For example, if the call is not a direct child of an statement expression or assignment statement
+    // then it's not reliable to determine if await is missing.
+    // User might return coroutine object from a function or pass it as an argument
+    if !possibly_missing_await(call, checker.semantic()) {
+        return;
+    }
+
+    let is_awaitable = is_awaitable_from_asyncio(call.func.as_ref(), checker.semantic())
+        || is_awaitable_func(call.func.as_ref(), checker.semantic());
+
+    // If call does not originate from asyncio or is not an async function, then it's not awaitable
+    if !is_awaitable {
+        return;
+    }
+
+    let mut diagnostic = Diagnostic::new(MissingAwaitForCoroutine {}, call.range);
```

Comment:
The curly braces aren't required for empty structs

```suggestion
    let mut diagnostic = Diagnostic::new(MissingAwaitForCoroutine, call.range);
```